scripts/memory.js

```diff
@@ -0,0 +1,45 @@
+class Memory {
+    constructor(key = "memory") {
+        this.key = key;
+        this.memoryValue = this.loadMemory();
+    }
+
+    loadMemory() {
+        try {
+            return parseFloat(localStorage.getItem(this.key)) || 0;
+        } catch (err) {
+            return 0;
+        }
+    }
+
+    // Store a new value in memory
+    memoryStore(value) {
+        this.memoryValue = value;
+        localStorage.setItem(this.key, this.memoryValue);
+    }
+
+    // Recall the stored value
+    memoryRecall() {
+        return this.memoryValue;
+    }
+
+    // Add a value to the stored memory
+    memoryAdd(value) {
+        this.memoryValue += value;
+        localStorage.setItem(this.key, this.memoryValue);
+    }
+
+    // Subtract a value from the stored memory
+    memorySubtract(value) {
+        this.memoryValue -= value;
+        localStorage.setItem(this.key, this.memoryValue);
+    }
```

Comment:
Same for here as well.
```suggestion
        if (Number.isFinite(value)) {
        this.memoryValue -= value;
        localStorage.setItem(this.key, this.memoryValue);
    }
```

Reply:
Added validation before assignment(https://github.com/krushna-simform/scientific-calculator/pull/4/commits/96c611513e8ca15533b3b50ac24d4880f92e1e11)